### PR TITLE
Feature: A new way to subscribe

### LINKED ISSUE #568: [Feature Request] Add/remove channel to profile from channel/video page
**Is your feature request related to a usage problem (not a bug)? Please describe.**
When I subscribe to a channel, I often want to add it to a profile immediately, so it's quite annoying you must go into the profile setting, look for the channel in the list, to finally add it to **one** profile.

**Describe the solution you'd like to see implemented**
There should be a dropdown list next to the subscribe button, to select in which profile(s) this channel is included.

### LINKED ISSUE #1615: A new way of subscribe using profiles
**Is your feature request related to a usage problem (not a bug)? Please describe.**
I find it a bit counter-intuitive to subscribe using multiple profiles.
My path to subscribe to a channel consists in searching for a channel, clicking on the channel, and hitting "subscribe".
The way FreeTube is right now, leads me to end up not using the multiples profiles, because I have to go to the top-right and change profile, and then clicking subscribe, this breaks the flow of adding a subscription.

**Describe the solution you'd like to see implemented**
I think a good solution is to have a kind of a dropdown combined with the subscribe button so when the user clicks the 'subscribe' button, he can select multiples profiles which to add the channel.

**Describe alternatives you've considered**

**Screenshots**
A quick rough sketch of what it may looks like
A rough sketch

**Additional context**

### PR DESCRIPTION
**Pull request by minhhang107**
---
New way to subscribe
---

**Pull Request Type**
Please select what type of pull request this is:
- [ ] Bugfix
- [x] Feature Implementation

**Related issue**
Closes #1615 
Closes #568

**Description**
Added ft-subscribe-button component
- The component includes the existing `Subscribe` button and a `plus` button
- The `plus` button allows users to to select the profile they want to add the subscription to
- Replace the existing `Subscribe` button with the new component in video page and channel page
- If users try to add subscription to the same profile twice, show a toast message

**Screenshots (if appropriate)**
Before:

before-video

before-channel

After:
after-video

after-channel

**Testing (for code that is not small enough to be easily understandable)**
Has this pull request been tested?
Yes

Please describe shortly how you tested it and whether there are any ramifications remaining:
Tested locally on dev by adding/removing channels to/from different profiles using the existing `Subscribe` button and the new button, tested on both video page and channel page

**Desktop (please complete the following information):**
 - OS: Windows
 - OS Version: 10
 - FreeTube version: 0.15.1

### CODE REVIEW

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@PrestonN idea time!

Maybe we can also do something like this when the playlist is implemented. So u can save a video in playlist in profile x while being in profile y

**Comment by minhhang107** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc The same thing can be done with playlist. Users can also add a video to multiple playlists in another profile. So what do you think about this PR? Are there anything that need to be fixed?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
I didn't had time to test in depth but initial thoughts about it that it is pretty well implemented

Would like to see #1833 mergerd before this one

File: src/renderer/components/ft-subscribe-button/ft-subscribe-button.vue [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  <div>
+    <div class="buttons">
+      <ft-button
+        :label="subscribedText"
+        class="subscribeButton"
+        background-color="var(--primary-color)"
+        text-color="var(--text-with-main-color)"
+        @click="handleSubscription"
+      />
+      <ft-button
```
Comment (ChunkyProgrammer) [inline src/renderer/components/ft-subscribe-button/ft-subscribe-button.vue:11]: An aria-label or title attribute should be added to this element for accessibility reasons

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> I didn't had time to test in depth but initial thoughts about it that it is pretty well implemented
> 
> Would like to see #1833 merged before this one

#1833 has been merged. Please address the changes that have been requested and resolve the merge conflicts in order for us to review.

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — CHANGES_REQUESTED** [review]
- I cant unsubscribe both on the video page and channel page. The same is for subscribing.

https://user-images.githubusercontent.com/73130443/169809748-355822fb-50fd-4737-a635-063996402700.mp4

- There is no dynamic way to unsubscribe now. I would like to see both a '-' and a '+' button . If u click on the '-' u'll see all profiles that have been subscribed to this channel and the ability to remove them. If u click on the '+' u'll get all the profiles that aren't subscribed to the channel and the ability to subscribe to the channel by clicking on the profile icon. So that means when i subscribe to a channel using the '+' list it will remove itself from the '+' list and add itself to the '-' list.

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — CHANGES_REQUESTED** [review]
Im sorry @ChunkyProgrammer for this big spam

https://user-images.githubusercontent.com/73130443/170069147-dfc3e4b7-d8df-4c7f-b78d-47566b0f9f52.mp4


https://user-images.githubusercontent.com/73130443/170069153-ebb2da94-53dc-4e0c-b1cc-b06b57059326.mp4


https://user-images.githubusercontent.com/73130443/170069155-bbd633ab-61d3-4287-8de1-e050ee05ab90.mp4


https://user-images.githubusercontent.com/73130443/170069160-36bbf33a-c379-4222-909d-8305cdc58c8d.mp4


https://user-images.githubusercontent.com/73130443/170069161-cbcb2096-4643-498a-b35e-d3b09ae0f67e.mp4


https://user-images.githubusercontent.com/73130443/170069163-a6162c10-3d06-4402-838c-8fe23179b88d.mp4

spacing needs to be the same
dropdown needs more padding
weirdWhiteSquare

**Comment by ChunkyProgrammer** [comment]
Things to fix:
Channel grid takes up 5 or 4

Not sure:
Dropdown obscures other elements: I think the dropdown is supposed to obscure the other elements?

Not to fix:
"Unsubscribe functionality doesn't match original functionality":  this is actually the intended functionality. There is an error with the code in the dev branch that prevents this from happening (it should be activeProfile._id instead of activeProfile) https://github.com/FreeTubeApp/FreeTube/blob/a5585afa18314cd4757b4a63a47c21cb3a9afc02/src/renderer/views/Channel/Channel.js#L563

**Comment by ChunkyProgrammer** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc could you test the grid thing again?

**Comment by ChunkyProgrammer** [comment]
> > @efb4f5ff-1298-471a-8973-3d47447115dc could you test the grid thing again? (i didn't reproduce it)
> 
> @ChunkyProgrammer I have a big brain fart. Which video/picture we talking about?

This one
https://user-images.githubusercontent.com/73130443/170069153-ebb2da94-53dc-4e0c-b1cc-b06b57059326.mp4

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> channel.with.same.name.cant.add.channel.mp4

Fixed
> Channelspace.moves.from.5.to.4.mp4

Fixed
> dropdown.closes.everytime.u.add.channel.mp4

Fixed
> dropdown.obscures.other.elements.mp4


Not sure if we should fix this. More discussion needed.
> unsubscribe.functionality.doesnt.match.the.original.functionality.mp4

unrelated to this PR

> wrong.icon.mp4

fixed
> 
> spacing needs to be the same 

Fixed
>dropdown needs more padding 

Fixed
>weirdWhiteSquare

This isnt fixed and didnt explain it well.

https://user-images.githubusercontent.com/73130443/171063698-516329ae-c0fb-4f58-893e-6d23558feeca.mp4

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@ChunkyProgrammer do you think that this closes https://github.com/FreeTubeApp/FreeTube/issues/2213?

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — DISMISSED** [review]
LGTM!

File: src/renderer/components/ft-subscribe-button/ft-subscribe-button.vue [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        text-color="var(--text-with-main-color)"
+        @click="showProfileList('unsubscribed')"
+      >
+        <font-awesome-icon
+          icon="minus"
+        />
+      </ft-button>
+    </div>
+    <div
+      v-if="showProfiles==='subscribed'"
```
Comment (PikachuEXE) [inline src/renderer/components/ft-subscribe-button/ft-subscribe-button.vue:37]: I think it becomes much easier to read when there is spacing
```suggestion
      v-if="showProfiles === 'subscribed'"
```

File: src/renderer/components/ft-subscribe-button/ft-subscribe-button.vue [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            :title="profile.name"
+            :aria-label="profile.name"
+          >
+            {{ notSubscribedProfileInitials[index] }}
+          </div>
+        </div>
+      </div>
+    </div>
+    <div
+      v-if="showProfiles==='unsubscribed'"
```
Comment (PikachuEXE) [inline src/renderer/components/ft-subscribe-button/ft-subscribe-button.vue:63]:
```suggestion
      v-if="showProfiles === 'unsubscribed'"
```

**Review by PikachuEXE — CHANGES_REQUESTED** [review]
Code side only small changes required
Feature review to be done later...

**Comment by PikachuEXE** [comment]
We do need some document to describe this new feature
Also the tooltips need to be shown faster

### Details
I see empty box if I have no other profiles... (and already subscribed)
I know there is tooltip but they show a few seconds later and I don't think many people would notice it
(It's still fine as is given it's explained well)
image

Same for profile full name tooltips
image

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> I see empty box if I have no other profiles... (and already subscribed)

After looking at this again i do agree it is a bit weird when u see this for the first time. Maybe we can put a text in the box.

For when u click on the + button 'All profiles are subscribed to this channel. If u want to remove this channel from a profile click on - button' and the opposite text for when u click on the - button.

We could also disable the button when there is nothing more to add and show a toast message. Only giving this option because the first option i gave could be problematic with CSS on smaller screens

**Comment by PikachuEXE** [comment]
I am not sure if this only requires a better documentation to explain it (plus the improvements of course)
Or also an option to only show these buttons when the new option is enabled (kind of advanced feature, with link to documentation)

**Comment by PikachuEXE** [comment]
My points still stand

**Comment by ChunkyProgrammer** [comment]
> My points still stand

I only resolved merge conflicts, didn't do any fixing

**Comment by PikachuEXE** [comment]
@ChunkyProgrammer 
I know ar~
Only for the record (I got goldfish memory)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> Only for the record (I got goldfish memory)

A @PikachuEXE  that behaves like Misty's Psyduck

**Comment by toby63** [comment]
One question: will or is this also possible in the playlist menu?
Right now I have a large number of videos saved in the playlist on profile x and I would like to be able to transfer some of the videos to other profiles, without clicking on them first.

Also a way to choose multiple videos and "copy" them to another profile would be nice.

**Comment by ChunkyProgrammer** [comment]
Outstanding issues: (copying for easier reference as github is hiding this by default)
We do need some document to describe this new feature
Also the tooltips need to be shown faster

### Details
I see empty box if I have no other profiles... (and already subscribed)
I know there is tooltip but they show a few seconds later and I don't think many people would notice it
(It's still fine as is given it's explained well)
image

Same for profile full name tooltips
image

**Comment by ChunkyProgrammer** [comment]
I believe I've addressed the concerns

**Comment by PikachuEXE** [comment]
Is it possible to disable the new buttons (plus & minus) by default?
I think we need better doc before we make it enabled by default

**Comment by ChunkyProgrammer** [comment]
@PikachuEXE what do you think needs to be documented in order to get this implemented?

**Comment by PikachuEXE** [comment]
1. Where do those new buttons do (and any subsequent events like press profile to add to that profile etc.)
2. When and where do they appear (they don't appear in subscribed channels view) (Do they still appear always or only in some conditions? I haven't tested again since last review. But it's unclear without testing which is why it should be documented)

**Comment by ChunkyProgrammer** [comment]
> 1. Where do those new buttons do (and any subsequent events like press profile to add to that profile etc.)
> 2. When and where do they appear (they don't appear in subscribed channels view) (Do they still appear always or only in some conditions? I haven't tested again since last review. But it's unclear without testing which is why it should be documented)

1) any suggestions on how to more clearly indicate that?
2) it's now a part of the SubscribedChannel view (this component should replace all subscribe buttons)

**Comment by ChunkyProgrammer** [comment]
IIRC, @absidue you said you'd like to implement this differently. Do you think this PR should be updated to just be a subscribe button component so we don't have to repeat the code + logic>

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Closing in favor of https://github.com/FreeTubeApp/FreeTube/pull/4238

**Pull request closed without merging**